### PR TITLE
FE-1187 add delete state to pill component

### PR DESCRIPTION
**Pull request by edleeks87**
# Description
Adds support for a 'close' Icon in the Pill component
Adds some specs to test new functionality

# Screenshots

- With no prop passed
screen shot 2019-01-15 at 14 31 30

- With prop passed and no cursor hover on icon
screen shot 2019-01-15 at 14 31 07

- With prop passed and cursor hover over icon
screen shot 2019-01-15 at 14 31 15

### CODE REVIEW

File: src/components/pill/pill.scss [hunk 1]
```diff
@@ -3,14 +3,38 @@
...
+.carbon-pill__delete-icon:before {
+  font: {
+    size: 10px;
+  }
+  line-height: 10px;
+  margin-left: 2px;
+  margin-right: 2px;
+}
+
+.carbon-pill__delete-icon:hover {
```
Comment (ianoxley) [inline src/components/pill/pill.scss]: Needs `:focus` styles too for keyboard users.

File: src/components/pill/pill.js [hunk 2]
```diff
@@ -92,6 +102,22 @@ class Pill extends React.Component {
...
+    if (this.props.onDelete) {
+      return (
+        <Icon
+          className='carbon-pill__delete-icon'
+          data-element='close'
+          onClick={ this.props.onDelete }
+          type='cross'
+          onBlur={ this.onCloseIconBlur }
+          bgSize='small'
+        />
```
Comment (ianoxley) [inline src/components/pill/pill.js]: Currently this won't be able to receive keyboard focus, so keyboard users won't be able to use the close icon.

You could wrap it in a `<button>` tag, which will add the correct behaviour for free (bar some CSS updates perhaps), or add an ARIA `role` attribute, `tabIndex` attribute, and event handlers to handle keyboard interaction.

File: src/components/pill/pill.scss (hunk 1 above)
Comment (edleeks87) [reply src/components/pill/pill.scss]: Thanks, I removed it when UX told me they wanted to go in a different direction than having the Icon tabbable, but obviously I should of left the css the same. I've changed it back to have `:focus` styles

File: src/components/pill/pill.js (hunk 2 above)
Comment (edleeks87) [reply src/components/pill/pill.js]: UX said they were thinking of going in a different direction than having the Icon tabbable so I removed the support for it. I've added it back for now as its something that can easily be removed when it's decided how they want it implemented

File: src/components/pill/pill.scss [hunk 3]
```diff
@@ -3,14 +3,46 @@
...
+  margin-right: -5px;
+}
+.carbon-pill__delete-icon:before {
+  font: {
+    size: 10px;
+  }
+  line-height: 10px;
+  margin-left: 2px;
+  margin-right: 2px;
+  position: relative; top: -1px;
```
Comment (harpalsingh) [inline src/components/pill/pill.scss]: should put top on a new line

File: src/components/pill/pill.scss [hunk 4]
```diff
@@ -3,14 +3,46 @@
...
+  font-size: 14px;
+  line-height: 14px;
+}
+
+.carbon-pill__delete-icon {
+  margin-left: 5px;
+  margin-right: -5px;
+}
+.carbon-pill__delete-icon:before {
+  font: {
```
Comment (harpalsingh) [inline src/components/pill/pill.scss]: can this just be font-size?

File: src/components/pill/pill.scss [hunk 5]
```diff
@@ -3,14 +3,46 @@
...
+.carbon-pill__delete-icon:hover {
+    box-shadow: 0 0 0 2px $blue-dark;
+    background-color: $blue-dark;
+    border-radius: 0 9px 9px 0;
+    color: #ffffff;
+}
+
+.carbon-pill__delete-icon:focus {
+  box-shadow: 0 0 0 2px $blue-dark;
+  background-color: $blue-dark;
```
Comment (harpalsingh) [inline src/components/pill/pill.scss]: alpha order?

Comment (edleeks87) [reply src/components/pill/pill.scss]: done

File: src/components/pill/pill.scss (hunk 4 above)
Comment (edleeks87) [reply src/components/pill/pill.scss]: Yeah I realised that after I pushed it, have changed it

File: src/components/pill/pill.scss (hunk 3 above)
Comment (edleeks87) [reply src/components/pill/pill.scss]: Done

**Comment by andrewjtait** [comment]
Will merge this and QA can be finished off as part of the Select/Option QA.

**Pull request merged**